taipy/gui/hooks.py

```diff
@@ -0,0 +1,37 @@
+import typing as t
+
+from taipy.logger._taipy_logger import _TaipyLogger
+
+from .utils.singleton import _Singleton
+
+
+class BaseGuiHook:
+    exposed_hooks: t.List[str] = []
+
+
+class GuiHooks(object, metaclass=_Singleton):
+    def __init__(self):
+        self.__hooks: t.List[BaseGuiHook] = []
+
+    def _register_gui_hook(self, hook: BaseGuiHook):
+        self.__hooks.append(hook)
+
+    def __getattr__(self, name: str):
+        def _resolve_hook(*args, **kwargs):
+            for hook in self.__hooks:
+                if name not in hook.exposed_hooks:
+                    continue
+                # call hook
+                try:
+                    res = getattr(hook, name)(*args, **kwargs)
```

Comment:
Check that this attribute is a function?

Reply:
That is an excellent point! thanks!!!